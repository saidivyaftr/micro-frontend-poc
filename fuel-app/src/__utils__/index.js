import React from 'react'
import { Provider } from 'react-redux'
import store from 'src/redux/Store'
import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store'

export const shallowWithMock = (
  Component,
  initialStoreState,
  initialProps = {},
) => {
  React.useLayoutEffect = React.useEffect
  let mockStore = configureStore()
  mockStore = mockStore(initialStoreState)
  return shallow(
    <Provider store={initialStoreState ? mockStore : store}>
      <Component {...initialProps} />
    </Provider>,
  )
}

export const mountWithMock = (
  Component,
  initialStoreState,
  initialProps = {},
) => {
  React.useLayoutEffect = React.useEffect
  let mockStore = configureStore()
  mockStore = mockStore(initialStoreState)
  return mount(
    <Provider store={initialStoreState ? mockStore : store}>
      <Component {...initialProps} />
    </Provider>,
  )
}
export const mountWithRouterMock = (
  Component,
  initialStoreState,
  initialProps = {},
) => {
  React.useLayoutEffect = React.useEffect
  let mockStore = configureStore()
  mockStore = mockStore(initialStoreState)
  return mount(
    <Provider store={initialStoreState ? mockStore : store}>
      <Component {...initialProps} />
    </Provider>,
  )
}
