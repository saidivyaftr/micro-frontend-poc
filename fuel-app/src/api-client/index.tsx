import axios from 'axios'
import { axiosWithCache } from './CacheClient'
import getConfig from 'next/config'
import paymentAPIs from './payments'
import statementAPIs from './statements'
import profileAPIs from './profile'
import { registerAPI } from './registerAPI'
const config = getConfig()
import boboAPIs from './bobo'
import serviceOrdersAPIs from './service-orders'
import selfServiceAPIs from './selfService'
import contactVerificationAPIs from './contact-verification'
import accountAPIs from './account'
import forgotAPIs from './forgot'
import { EQUIPMENT_PAYLOAD } from 'src/libs/returns/ReturnsReview'

const publicRuntimeConfig = config?.publicRuntimeConfig

// Returns BASE URL based upon the env
export const getBaseURL = () => {
  const origin = window.location.origin
  const allowedHosts = ['localhost.frontier.com:3000']
  const nextBasePath =
    publicRuntimeConfig?.basePath?.replace(/\/+/g, '/') || 'pages'
  if (!origin) {
    return 'https://www.frontier.com/' + nextBasePath
  }
  if (process.env.NODE_ENV === 'production') {
    return origin.includes(nextBasePath)
      ? trimOrigin(origin)
      : origin + nextBasePath
  }
  if (allowedHosts.includes(window.location.host)) {
    return `${origin}/${nextBasePath.replace(/\/+/g, '')}`
  }
  return 'http://localhost:3000/' + nextBasePath.replace(/\/+/g, '')
}

const trimOrigin = (url = '') => {
  if (url && url[url.length - 1] === '/') {
    return url.slice(0, -1)
  }
  return url
}

// API client methods
const client = {
  // Importing other methods
  ...registerAPI,
  ...boboAPIs(getBaseURL),
  ...serviceOrdersAPIs(getBaseURL),
  ...selfServiceAPIs(getBaseURL),
  ...contactVerificationAPIs(getBaseURL),
  ...accountAPIs(getBaseURL),
  ...forgotAPIs(getBaseURL),

  // Universal methods
  profile: () => {
    return axios.get(`${getBaseURL()}/api/profile`)
  },
  session: () => {
    return axios.get(`${getBaseURL()}/api/session`)
  },
  postSession: (data: any) => {
    return axios.post(`${getBaseURL()}/api/session`, data)
  },
  getAddressDetails: (environment: string, controlNumber: string) => {
    return axios.get(
      `${getBaseURL()}/api/address?environment=${environment}&controlNumber=${controlNumber}`,
    )
  },
  getPredictiveSuggestions: (address: string, inFootPrint = false) => {
    return axios.get(
      `${getBaseURL()}/api/address-search?address=${address}&inFootPrint=${inFootPrint}`,
    )
  },
  serviceabilityCheck: (env: string, controlNumber: string) => {
    return axios.get(
      `${getBaseURL()}/api/serviceability?env=${env}&controlNumber=${controlNumber}`,
    )
  },
  chatAPI: () => {
    console.log('step3::::')
    // return axios.post(
    //   `${getBaseURL()}/api/chatAPI`,
    // )

    return axios.post(`${getBaseURL()}/api/chatAPI`, {})
  },
  keepAlive: () => {
    return axios.head(`${getBaseURL}/api/keepAlive`)
  },
  submitACPForm: (data: any) => {
    return axios({
      url: `${getBaseURL()}/api/acp-form`,
      method: 'POST',
      data,
      headers: {
        'requesting-application': 'ResiEcomm',
      },
    })
  },
  certifyInternetUsage: (data: any) => {
    return axios({
      url: `${getBaseURL()}/api/certify-internet-usage`,
      method: 'PUT',
      data,
      headers: {
        'requesting-application': 'ResiEcomm',
      },
    })
  },
  getUserTosStatuses: (accountUuid: string) => {
    return axios({
      url: `${getBaseURL()}/api/user-tos-status?uuid=${accountUuid}`,
      method: 'GET',
      headers: {
        'requesting-application': 'ResiEcomm',
      },
    })
  },
  updateTosStatus: (accountUuid: any, data: any) => {
    return axios({
      url: `${getBaseURL()}/api/update-tos-status?uuid=${accountUuid}`,
      method: 'PUT',
      data,
      headers: {
        'requesting-application': 'ResiEcomm',
      },
    })
  },
  login: (data: any) => {
    return axios({
      url: `${getBaseURL()}/api/login`,
      method: 'POST',
      data,
    })
  },
  checkOutages: (phone: string) => {
    return axios.get(`${getBaseURL()}/api/checkoutages?phone=${phone}`)
  },
  ...statementAPIs(getBaseURL),
  ...paymentAPIs(getBaseURL),
  ...profileAPIs(getBaseURL),
  createPegaInteraction: (payload: any) => {
    return axios.post(`${getBaseURL()}/api/pega`, payload)
  },
  ccpaSignIn: (payload: any) => {
    return axios.post(`${getBaseURL()}/api/ccpa/signin`, payload)
  },
  ccpaReview: (token: any) => {
    return axios.post(`${getBaseURL()}/api/ccpa/review-request`, { token })
  },
  ccpaDelete: (token: any) => {
    return axios.post(`${getBaseURL()}/api/ccpa/delete-request`, { token })
  },
  ccpaReviewDelete: (token: any) => {
    return axios.post(`${getBaseURL()}/api/ccpa/review-and-delete-request`, {
      token,
    })
  },
  postPaperlessBillPreference: (accountId: string, payload: any) => {
    return axios.post(
      `${getBaseURL()}/api/accounts/${accountId}/paperlessBilling`,
      payload,
    )
  },
  postEvent: (payload: any) => {
    return axios.post(`${getBaseURL()}/api/postevent`, payload)
  },
  serviceabilityUpdatedCheck: (env: string, controlNumber: string) => {
    return axios.get(
      `${getBaseURL()}/api/serviceabilityUpdated?env=${env}&controlNumber=${controlNumber}`,
    )
  },
  getAccounts: (forceFetch = false) => {
    return axiosWithCache(`${getBaseURL()}/api/accounts`, undefined, forceFetch)
  },
  getConstantsByLang: (language: string, serviceType: string) => {
    return axios.get(`${getBaseURL()}api/constants/${language}/${serviceType}`)
  },
  getProducts: (accountId: string, detailType: string) => {
    return axios.get(
      `${getBaseURL()}/api/accounts/${accountId}/products/${detailType}`,
    )
  },
  carrierFreeze: (payload: any) => {
    return axios.post(`${getBaseURL()}/api/ploc_opt_out`, payload)
  },
  tpVerify: (payload: any) => {
    return axios.post(
      `${getBaseURL()}/api/serviceOrders/verificationSearch`,
      payload,
    )
  },
  cpniPost: (payload: any) => {
    return axios.post(`${getBaseURL()}/api/cpni_opt_out`, payload)
  },
  tpvOrderConfirmation: (payload: any, serviceId: string) => {
    return axios.post(
      `${getBaseURL()}/api/serviceOrders/${serviceId}/confirmationOrder`,
      payload,
    )
  },
  getOutages: () => {
    return axios.get(`${getBaseURL()}/api/outages`)
  },
  getDisaster: () => {
    return axios.get(`${getBaseURL()}/api/disasters`)
  },
  getServiceOutages: () => {
    return axios.get(`${getBaseURL()}/api/serviceOutages`)
  },
  quickLinksMetaData: () => {
    return axios.get(`${getBaseURL()}/api/quickLinks-metadata`)
  },
  getVoicemail: () => {
    return axios.get(`${getBaseURL()}/api/accessnumber/voicemail`)
  },
  getNotifications: (accountId: string) => {
    return axios.get(
      `${getBaseURL()}/api/accounts/${accountId}/notification-preferences`,
    )
  },
  getCCPAReviews: (accountId: string) => {
    return axios.get(`${getBaseURL()}/api/accounts/${accountId}/ccpa/reviews`)
  },
  postNotificationSettings: (accountId: string, payload: any) => {
    return axios.post(
      `${getBaseURL()}/api/accounts/${accountId}/notification-preferences`,
      payload,
    )
  },
  updateNotification: (accountId: string, payload: any) => {
    return axios.patch(
      `${getBaseURL()}/api/accounts/${accountId}/notification-preferences`,
      payload,
    )
  },
  getIspEmails: (serviceId: string) => {
    return axios.get(
      `${getBaseURL()}/api/services/${serviceId}/isp/emailAccounts`,
    )
  },
  multifamily: (payload: any) => {
    return axios.post(`${getBaseURL()}/api/multifamily`, payload)
  },
  logout: () => {
    return axios.post(`${getBaseURL()}/api/logout`, {})
  },
  getServicesCached: (
    includeDetails?: boolean,
    ignoreAuthErrors?: boolean,
    forceFetch?: boolean,
  ) => {
    const ignoreAuthErrorsParams = ignoreAuthErrors
      ? { ignoreAuthErrors }
      : undefined
    const includeDetailsParams = includeDetails ? { includeDetails } : undefined
    return axiosWithCache(
      `${getBaseURL()}/api/services`,
      {
        params: {
          ...ignoreAuthErrorsParams,
          ...includeDetailsParams,
        },
      },
      forceFetch,
    )
  },
  getServices: (includeDetails?: boolean, ignoreAuthErrors?: boolean) => {
    const ignoreAuthErrorsParams = ignoreAuthErrors
      ? { ignoreAuthErrors }
      : undefined
    const includeDetailsParams = includeDetails ? { includeDetails } : undefined
    return axios.get(`${getBaseURL()}/api/services`, {
      params: {
        ...ignoreAuthErrorsParams,
        ...includeDetailsParams,
      },
    })
  },
  getUPSReturns: (token?: string) => {
    const tokenParams = token ? { token } : undefined
    return axios.get(`${getBaseURL()}/api/upsReturns`, {
      params: {
        ...tokenParams,
      },
    })
  },

  getAccountDetails: (accountId: string) => {
    return axios.get(`${getBaseURL()}/api/accounts/${accountId}`)
  },
  getEquipments: (accountId: string, payload: any) => {
    return axios.post(
      `${getBaseURL()}/api/accounts/${accountId}/equipments`,
      payload,
    )
  },
  changePassword: (oldPassword: string, newPassword: string) => {
    return axios.post(`${getBaseURL()}/api/profile`, {
      changePassword: true,
      oldPassword,
      newPassword,
    })
  },
  changeProfileEmail: (email: string) => {
    return axios.post(`${getBaseURL()}/api/profile`, {
      changeProfileEmail: true,
      email,
    })
  },
  equipmentsFindGet: (guestId: string) => {
    return axios.get(
      `${getBaseURL()}/api/services/${guestId}/returnEquipmentMailerOrders`,
    )
  },
  equipmentsFindPost: (guestId: string, payload: EQUIPMENT_PAYLOAD) => {
    return axios.post(
      `${getBaseURL()}/api/services/${guestId}/returnEquipmentMailerOrders`,
      payload,
    )
  },
  getCompleteRegistration: (token: string) => {
    return axios.get(`${getBaseURL()}/api/legacy/completeRegistration/${token}`)
  },
  postCompleteRegistration: (token: string, payload: any) => {
    return axios.post(
      `${getBaseURL()}/api/legacy/completeRegistration/${token}`,
      payload,
    )
  },
  resendRegistrationEmail: (payload: any) => {
    return axios.post(
      `${getBaseURL()}/api/legacy/resendRegistrationEmail`,
      payload,
    )
  },
  getChatAccessCode: () =>
    axiosWithCache(`${getBaseURL()}/api/authorizationcode`),
}
export default client
