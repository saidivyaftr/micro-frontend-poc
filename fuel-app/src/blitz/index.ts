import Typography from './components/Typography'
import Button from './components/Button'
import Accordion from './components/Accordion'
import FourTiles from './components/FourTiles'
import Hero from './components/Hero'
import CheckAvailability from './components/CheckAvailability'
import Calendar from './components/Calendar'
import Header from './components/Header'
import Footer from './components/Footer'
import DarkFooter from './components/DarkFooter'
import TwoColumnLayout from './components/TwoColumnLayout'
import SwiperQuotes from './components/SwiperQuotes'
import ImagePerk from './components/ImagePerk'
import Stripes from './components/Stripes'
import ComparisonTable from './components/ComparisonTable'
import TwoCards from './components/TwoCards'
import Alert from './components/Alert'
import Video from './components/Video'
import Modal from './components/Modal'
import Loading from './components/Loading'
import Hamburger from './components/Hamburger'
import InjectHTML from './components/InjectHTML'
import Tooltip from './components/Tooltip'
import TooltipPopover from './components/TooltipPopover'
import LinksAccordion from './components/LinksAccordion'
import UtilityNav from './components/UtilityNav'
import NavAccordion from './components/NavAccordion'
import NotificationBanner from './components/NotificationBanner'
import Tile from './components/Tile'
import CtaBanner from './components/CtaBanner'
import CardAndImage from './components/CardAndImage'
import RichText from './components/RichTextComponent'
import HoverReveal from './components/HoverReveal'
import HeroStripes from './components/HeroStripes'
import ArticleCard from './components/ArticleCard'
import SpeedCard from './components/SpeedCard'
import TitleWithCaption from './components/TitleWithCaption'
import TitleWithButton from './components/TitleWithButton'
import ProsAndConsTable from './components/ProsAndConsTable'
import ScrollTop from './components/ScrollTop'
import ButtonWithChatLink from './components/ButtonWithChatLink'
import InfoModal from './components/InfoModal'
import Picture from './components/Picture'
import SocialMediaLinks from './components/SocialMediaLinks'
import VideoCollection from './components/VideoCollection'
import Breadcrumb from './components/Breadcrumb'
import VideoModal from './components/VideoModal'
import OTPInput from './components/OTPInput'
import ToggleButton from './components/ToggleButton'
import Card, { CardWithTitle } from './components/Card'
import ArrowLink from './components/ArrowLink'
import StickyBanner from './components/StickyBanner'
import Checkbox from './components/Checkbox'
import TrophyCase from './components/TrophyCase'
import AwardCarousel from './components/AwardCarousel'
import PriceCard from './components/PriceCard'
import AwardCarouselWithCard from './components/AwardCarouselWithCard'
import BlackHeader from './components/BlackHeader'
import TrackOrders from './components/TrackOrders'
import Skeleton from './components/Skeleton'
import BlockTable from './components/BlockTable'
import Testimonial from './components/Testimonial'
export {
  TrackOrders,
  BlackHeader,
  Typography,
  Button,
  Accordion,
  FourTiles,
  Hero,
  TwoColumnLayout,
  Header,
  Footer,
  SwiperQuotes,
  ImagePerk,
  Stripes,
  ComparisonTable,
  TwoCards,
  Alert,
  Video,
  Modal,
  Loading,
  Hamburger,
  InjectHTML,
  Tooltip,
  TooltipPopover,
  LinksAccordion,
  UtilityNav,
  NavAccordion,
  CheckAvailability,
  Calendar,
  NotificationBanner,
  Tile,
  CtaBanner,
  CardAndImage,
  RichText,
  HoverReveal,
  HeroStripes,
  ArticleCard,
  SpeedCard,
  DarkFooter,
  TitleWithCaption,
  TitleWithButton,
  ProsAndConsTable,
  ScrollTop,
  ButtonWithChatLink,
  InfoModal,
  Picture,
  SocialMediaLinks,
  VideoCollection,
  Breadcrumb,
  VideoModal,
  OTPInput,
  ToggleButton,
  Card,
  CardWithTitle,
  ArrowLink,
  StickyBanner,
  Checkbox,
  TrophyCase,
  AwardCarousel,
  PriceCard,
  AwardCarouselWithCard,
  Skeleton,
  BlockTable,
  Testimonial,
}

export type { ITypography } from './components/Typography'
