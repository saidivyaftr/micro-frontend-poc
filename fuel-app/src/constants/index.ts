export * from './trackingPageNames'
export * from './clickEvents'
export * from './customerTypes'
export * from './serviceTypes'
export * from './storageKeys'
export * from './apiRoutes'
export * from './appRoutes'
export * from './layoutConfigs'
export * from './formMessages'
export * from './regex'
export * from './verbiage'
export * from './responseCodes'
export * from './mapDataTypes'
export * from './metaData'
export * from './timeDateRanges'
