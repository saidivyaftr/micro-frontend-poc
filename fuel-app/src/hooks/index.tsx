import useDebounce from './useDebounce'
import useAppData from './useAppData'
import useAddressPredictor from './useAddressPredictor'
import { useWindowDimensions } from './useWindowDimensions'
import usePageLoadEvents from './usePageLoadEvents'
import useChatState from './useChatState'
import useAlterChatRedirects from './useAlterChatRedirects'
import useAlterTermLinks from './useAlterTermLinks'
import useIntersection from './useIntersection'
import useUpdateSearchPlaceholder from './useUpdateSearchPlaceholder'
import useFormatterAccountNumber from './useFormatterAccountNumber'
import useSticky from './useSticky'
import useDelay from './useDelay'
import useIsLoadingFromApp from './useIsLoadingFromApp'
import useShowTerms from './useShowTerms'
import useCidParameter from './useCidParameter'
import useBreadcrumbParser from './useBreadcrumbParser'
import useAccountPageLoadEvents from './useAccountPageLoadEvents'
import useCurrentAccountName from './useCurrentAccountName'
import { useMainLayoutEffects } from './useMainLayoutEffects'

export {
  useDebounce,
  useAppData,
  useAddressPredictor,
  useWindowDimensions,
  usePageLoadEvents,
  useChatState,
  useShowTerms,
  useAlterChatRedirects,
  useAlterTermLinks,
  useIntersection,
  useUpdateSearchPlaceholder,
  useFormatterAccountNumber,
  useSticky,
  useDelay,
  useIsLoadingFromApp,
  useCidParameter,
  useBreadcrumbParser,
  useAccountPageLoadEvents,
  useCurrentAccountName,
  useMainLayoutEffects,
}
