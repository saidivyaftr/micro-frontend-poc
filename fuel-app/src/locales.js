export const locales = [
  { name: 'English', code: 'en', image: 'engIcon' },
  { name: 'Espa√±ol', code: 'es', image: 'yellIcon' },
]

export const TRANSLATIONS_ENABLED_PAGES = [
  '/verify',
  '/resources/verification-confirmation',
  '/discount-programs/california-teleconnect-fund',
  '/discount-programs/lifeline-program/california',
  '/discount-programs/lifeline-program/texas',
  '/discount-programs/lifeline-program/florida',
  '/resources/california-emergency',
  '/resources/email-disclaimer',
  '/corporate/policies',
  '/corporate/privacy-policy',
  '/corporate/terms',
  '/corporate/privacy-policy-california',
  '/california-privacy-law',
  '/helpcenter/topics/billing-faqs',
  // '/corporate/responsibility/annual-notice',
]
