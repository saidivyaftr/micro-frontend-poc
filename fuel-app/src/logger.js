import { createLogger, format, transports } from 'winston'
import 'winston-daily-rotate-file'
// import winston from 'winston/lib/winston/config'

const initializeLogger = () => {
  // commenting the logger file code for future reference
  // const fileLogTransport = new transports.DailyRotateFile({
  //   filename: `logs/${_fileName}-%DATE%.log`,
  //   datePattern: 'YYYY-MM-DD',
  //   zippedArchive: true,
  //   maxSize: '20m',
  //   maxFiles: '60d',
  // })
  const consoleTransport = new transports.Console({
    handleExceptions: false,
    json: false,
    colorize: true,
    raw: true,
    format: format.combine(
      format.colorize(),
      format.printf((i) => `${i.message}`),
    ),
  })
  const logger = createLogger({
    format: format.combine(
      format.timestamp({
        format: 'YYYY-MM-DD HH:mm:ss',
      }),
      format.errors({ stack: true }),

      format.splat(),
      format.printf(
        ({ level, message, label = process.env.NODE_ENV, timestamp }) =>
          `${timestamp} [${label}] ${level}: ${message}`,
      ),
    ),
    defaultMeta: { service: 'my-app' },
    transports: [consoleTransport],
  })
  // commenting the logger file code for future reference
  // if (process.env.NODE_ENV !== 'development') {
  //   logger.add(fileLogTransport)
  // }

  return logger
}

export default initializeLogger()
