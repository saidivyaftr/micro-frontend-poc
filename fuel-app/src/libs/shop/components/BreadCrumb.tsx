import { useAppData } from 'src/hooks'
import colors from '@/shared-ui/colors'
import { Breadcrumbs, Link, makeStyles } from '@material-ui/core'
import { InjectHTML } from '@/shared-ui/components'

const BreadCrumb = () => {
  const breadCrumbs = useAppData('breadCrumbs', true)
  const classes = useStyles()
  return (
    <Breadcrumbs className={classes.breadcrumbs} aria-label="breadcrumb">
      {breadCrumbs?.parentName?.value && (
        <Link
          href={`${breadCrumbs?.parentPath?.url}`}
          className={classes.boldText}
        >
          <InjectHTML
            fontType="boldFont"
            styleType="legal"
            data-testid="parentPageName"
            value={breadCrumbs?.parentName?.value}
          />
        </Link>
      )}
      {breadCrumbs?.name?.value && (
        <InjectHTML
          fontType="regularFont"
          styleType="legal"
          data-testid="currentPageName"
          value={breadCrumbs?.name?.value}
        />
      )}
    </Breadcrumbs>
  )
}

const useStyles = makeStyles(({}) => ({
  breadcrumbs: {
    textTransform: 'capitalize',
    maxWidth: '1200px',
    margin: 'auto',
    padding: '20px 16px',
    fontSize: '12px',
    color: colors.main.blackCoral,
    '& sup': {
      fontSize: '0.5em',
      position: 'relative',
      top: '-1px',
    },
  },
  boldText: {
    color: '#141928',
    '&:hover': {
      textDecoration: 'underline',
    },
  },
}))

export default BreadCrumb
