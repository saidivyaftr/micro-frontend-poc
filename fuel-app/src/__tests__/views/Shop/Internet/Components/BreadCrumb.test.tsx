import BreadCrumb from 'src/libs/shop/components/BreadCrumb'
import { mountWithMock } from 'src/__utils__'

describe('Bread Crumb', () => {
  it('should render correctly without store', () => {
    const component = mountWithMock(BreadCrumb)
    expect(component).toMatchSnapshot()
  })
  it('should render breadcrumb correctly with store', () => {
    const mockData = {
      breadCrumbs: {
        parentName: {
          value: 'Frontier® Fiber Internet',
        },
        parentPath: {
          url: '/shop/internet/fiber-optic-internet',
        },
        name: {
          value: 'Frontier® Fiber Internet Gig Service',
        },
      },
    }
    const wrapper = mountWithMock(BreadCrumb, {
      appData: mockData,
    })
    expect(wrapper).toMatchSnapshot()
  })
})
