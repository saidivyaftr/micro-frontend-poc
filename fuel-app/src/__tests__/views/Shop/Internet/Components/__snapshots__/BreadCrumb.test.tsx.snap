// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bread Crumb should render breadcrumb correctly with store 1`] = `ReactWrapper {}`;

exports[`Bread Crumb should render correctly without store 1`] = `ReactWrapper {}`;
